legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java

```diff
@@ -180,14 +192,27 @@ private void handleCursorRequest(final RestRequest request, final String cursor,
         cursorRestExecutor.execute(client, request.params(), channel);
     }
 
+    /**
+     * Log error message for exception and increment failure statistics.
+     * @param e : Caught exception.
+     */
     private static void logAndPublishMetrics(final Exception e) {
+        /**
+         * Use PrintWriter to copy the stack trace for logging. This is used to anonymize
+         * log messages, and can be reverted to the simpler implementation when
+         * the anonymizer is fixed.
+         */
+        StringWriter sw = new StringWriter();
+        e.printStackTrace(new PrintWriter(sw));
+        String stackTrace = sw.toString();
```

Comment:
Can't his block (205-207) be with line 215? It'll be easier to follow.

Reply:
Resolved under: [08981598f03df7da85f23fd547b70c1a08318c8c](https://github.com/opensearch-project/sql/pull/691/commits/08981598f03df7da85f23fd547b70c1a08318c8c)

Reply:
It is for debug purpose, right? could it be wrap inside if (log.debug)?

Reply:
We want the stack trace to be logged under normal debug levels as error. So not for debugging purposes only but error reporting like prior to this new implementation. I could wrap this inside of a Log.Error if that is more of your liking.